src/pages/main-page/pages/Content.jsx

```diff
@@ -1,10 +1,57 @@
-import IdolListTabContent from '../components/Idol-list-tab-content';
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useChartPagination } from '@hooks/useChartPagination';
+import Button from '@components/common/Button';
+import IdolListTabContent from '../components/IdolListTabContent';
+import IdolList from '@components/vote/IdolList';
 
+/**
+
+* 이들의 차트 콘텐츠 페이지
+* - 성별(gender)에 따라 차트 데이터를 요청해 리스트로 출력
+* - 커서 기반 페이지네이션으로 "더보기" 구현
+*/
 export default function Content() {
+  // 현재 URL 경로에 따라 성별 결정
+  const location = useLocation();
+  const gender = location.pathname.includes('female') ? 'female' : 'male';
+
+  // 차트 페이징 관련 훅 (데이터, 로딩 상태, 요청 함수 등)
+  const {
+    chartDataList,
+    nextCursor,
+    isLoading,
+    fetchIdolData,
+    resetPagination,
+  } = useChartPagination(gender);
+
+  // gender가 바뀌면 초기화 후 첫 페이지 데이터를 재요청
+  useEffect(() => {
+    resetPagination();
+    fetchIdolData(0);
+  }, [gender]);

```

Comment:
현재 `useEffect`에서 `fetchIdolData(0)`을 호출하는데, 이미 초기화된 데이터가 있을 경우 중복 호출을 피하는 로직을 추가할 수 있습니다.

```suggestion
useEffect(() => {
    if (chartDataList.length === 0) { // 데이터가 없다면만 요청
      resetPagination();
      fetchIdolData(0);
    }
  }, [gender, chartDataList]);

```

Reply:
성별 탭 전환 시마다 데이터를 새로 불러오는 흐름을 명확히 하기 위해 gender만 의존성으로 유지했습니다!